compiler/hie-core/src/Development/IDE/LSP/Server.hs

```diff
@@ -33,6 +36,21 @@ import qualified Language.Haskell.LSP.Control as LSP
 import qualified Language.Haskell.LSP.Core as LSP
 import qualified Language.Haskell.LSP.Messages as LSP
 import qualified Language.Haskell.LSP.Types as LSP
+import Development.IDE.Core.Service
+
+
+data RunHandler = RunHandler
+    {addResponse :: forall m req resp . (ResponseMessage resp -> LSP.FromServerMessage) -> (IdeState -> req -> IO resp) -> Maybe (LSP.Handler (RequestMessage m req resp))
+    ,addNotification :: forall m req . (IdeState -> req -> IO ()) -> Maybe (LSP.Handler (NotificationMessage m req))
```

Comment:
I don’t understand in what sense this “adds” anything. This seems more like a constructor function for building handlers. I also don’t understand the name `RunHandler`.

Reply:
The names aren't good. Let me have a new go.